interpreter/code_interpreters/dockerfiles/dev_note.md

```diff
@@ -0,0 +1,22 @@
+=======================================================================
+                              NOTE TO DEVELOPERS
+=======================================================================
+
+Please avoid manually editing the following files:
+- Dockerfile
+- requirements.txt
+- hash.json
+
+These files are key components for Open-Interpreter's containerized execution features. Manually editing them can disrupt the program's ability to:
+
+1. Know when to rebuild the Docker image.
+2. Perform other related functionalities efficiently.
+
+If you need to make adjustments, kindly use the 'DockerManager' class. It offers convenient methods like:
+- add_dependency
+- remove_dependency
+- add_language
+
+Your cooperation helps maintain a smooth and reliable development workflow.
```

Comment:
This seems a little unintuitive and would really benefit from concrete examples and explanations of the how, what, and why of this design choice.

But, I wonder if, given that Python has a `requirements.txt` and Node has a `package.json,`, it would make more sense to just leverage those files instead of putting everything into a `requirements.txt` and adding a layer of indirection and abstraction? I'm not sure if R manages dependencies similarly, but maybe we could implement something unique for that edge case instead.

Reply:
The addition of the ```DockerManager``` class was mainly aimed at providing a convienent interface to add and remove dependencies to the container, so that users / The Language model dont have to re-install the dependencies they want at runtime every time they want to use it. 

The reason I merged all of the dependency lists into one file served 3 main purposes.
 1.  just to avoid clutter.
 2.  so that the interface which ```DockerManager``` uses for  adding and removing dependencies could be standardized. 
 3. also, this way theres only one file to copy into the container when building the image which just makes image construction faster albeit by a small ammount. 

I also think I am going to remove the ```add_language``` method of the manager class.  the rationale behind this is because it would take a decent ammount of work in the rest of the codebase to truly support a new language; and at that point they may aswell just modify the Dockerfile and requirements themselves.